src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,130 @@
-import React from 'react';
+/* eslint-disable max-len */
+import classNames from 'classnames';
+import React, { useMemo, useState, useEffect } from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+type Props = {
+  todos: Todo[];
+  selectedUserId: number;
+  selectUser: (id: number) => void;
+};
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectedUserId,
+  selectUser,
+}) => {
+  const [query, setQuery] = useState<string>('');
+  const [statusFilter, setStatusFilter] = useState<string>('all');
+  const [random, setRandom] = useState<boolean>(false);
```

Comment:
You don't need to hold random in state, change todos order onClick on randomize button

Reply:
can`t delete 'random' state, the code crashes

Reply:
you don't need to keep a random variable with parameters true or false, you can do the previous comment by applying random selection sort(all, active, completed) every time you click on on randomize button. You can use object with number keys and order method in properties and choose random key from object or get random element from array like ['all', 'active', 'completed'] [hint](https://stackoverflow.com/questions/5915096/get-a-random-item-from-a-javascript-array)